js/client-chat.js

```diff
@@ -837,7 +837,10 @@
 			case 'rating':
 			case 'ladder':
 				if (app.localLadder) return text;
-				if (!target) target = app.user.get('userid');
+				if (!target) {
+					target = app.user.get('userid');
+					if (this.battle) target += ", " + this.id.split('-')[1];
+				}
```

Comment:
A good start, but this only works for `/rank` by itself, not for `/rank otherplayer`. I went ahead and implemented it for both.